### PR TITLE
Defines OSSL_SSIZE_MAX

### PR DESCRIPTION
**Pull request by RJPercival**
ossl_ssize_t varies in size, depending on the platform, so it is useful to also have its max value defined.

Updated bss_bio.c to use this, instead of defining SSIZE_MAX itself (and getting it wrong on 64-bit platforms that don't define SSIZE_MAX).

### CODE REVIEW

File: include/openssl/e_os2.h [hunk 1]
```diff
@@ -262,17 +262,21 @@ extern "C" {
 # ifdef _WIN32
 #  ifdef _WIN64
 #   define ossl_ssize_t __int64
+#   define OSSL_SSIZE_MAX _I64_MAX
 #  else
 #   define ossl_ssize_t int
+#   define OSSL_SSIZE_MAX INT_MAX
 #  endif
```
Comment (benlaurie) [inline include/openssl/e_os2.h]: Does windows not have a native ssize_t?

Comment (RJPercival) [inline include/openssl/e_os2.h]: It's a posix type, so it appears not. It does have SSIZE_T though (https://msdn.microsoft.com/en-us/library/windows/desktop/aa383751(v=vs.85).aspx#SSIZE_T). I think switching over to that would belong in a separate PR though.

**Comment by benlaurie** [comment]
+1

**Comment by levitte** [comment]
+1

Feel free to merge, @benlaurie

**Comment by kroeckx** [comment]
Labels set based on other's review, I did not review this.

**Comment by richsalz** [comment]
done with e634b448c3113545ff5f9a70e74a0ce694a00e77
